/*
  Device Management

  This BfB section holds messages used for device management.

  The version of the OpenAPI document: 2

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package com.bosch.bfb.protobuf.devicemanagement.v2;

import public "device_to_backend_message_meta.proto";

message CertificateReportEvent {

  DeviceToBackendMessageMeta meta = 3347973;

  // list of certificates and their respective purposes reported by the device
  repeated CertificateEntry certificates = 394850748;

}

message CertificateEntry {


  PurposeEnum purpose = 220463842;

  // X.509 certificate in PEM encoding (base64)
  string certificate = 341787034;

}
// Enum describing the purpose a certificate serves.
enum PurposeEnum {
  UNKNOWN_PURPOSE = 0;
  BFB_CLIENT_TLS_AUTH = 1;
  ARTIFACT_ENCRYPTION = 2;
}